compiler/packages/babel-plugin-react-compiler/src/__tests__/fixtures/compiler/dont-merge-if-dep-is-inner-declaration-of-previous-scope.expect.md

```diff
@@ -99,41 +99,41 @@ function Component(t0) {
     t2 = $[12];
   }
   let t3;
-  if ($[13] !== t2 || $[14] !== x) {
-    t3 = <ValidateMemoization inputs={t2} output={x} />;
-    $[13] = t2;
-    $[14] = x;
+  if ($[13] !== a || $[14] !== b) {
+    t3 = [a, b];
+    $[13] = a;
+    $[14] = b;
     $[15] = t3;
   } else {
     t3 = $[15];
   }
   let t4;
-  if ($[16] !== a || $[17] !== b) {
-    t4 = [a, b];
-    $[16] = a;
-    $[17] = b;
+  if ($[16] !== t2 || $[17] !== x) {
+    t4 = <ValidateMemoization inputs={t2} output={x} />;
+    $[16] = t2;
+    $[17] = x;
     $[18] = t4;
   } else {
     t4 = $[18];
   }
   let t5;
-  if ($[19] !== t4 || $[20] !== z) {
-    t5 = <ValidateMemoization inputs={t4} output={z} />;
-    $[19] = t4;
+  if ($[19] !== t3 || $[20] !== z) {
+    t5 = <ValidateMemoization inputs={t3} output={z} />;
+    $[19] = t3;
```

Comment:
this example would be more optimal if we could reorder the array literals too, and move them closer to the `<ValidateMemoization>`, but that gets complicated if the arrays could be mutated